p2p/discovery/discovery.go

```diff
@@ -0,0 +1,124 @@
+package discovery
```

Comment:
are we still doing discovery every 6 hours?

Reply:
No, we are not. Currently, discovery include three parts:
1. Basic DHT table discovery, this is natively implemented in libp2p `host.Start()`
2. PubSub topic discovery, When initializing pubsub (`node.StartPubSub()`), a new option `WithDiscovery` is added, so that we are using authenticated way of discover in pubsub.
3. Stream topic. This topic discovery is added in `streammanager` in the next PR.

Reply:
I remember back in 2019, if we don't do discovery every 6 hours, the node may forget the peers.  I don't know if it is still the case though.

Reply:
Yea, let's try it out in the testing network. :)